addons/test_mail_full/models/test_mail_models_mail.py

```diff
@@ -38,6 +38,16 @@ def _compute_access_url(self):
         for record in self.filtered('id'):
             record.access_url = '/my/test_portal_no_partner/%s' % self.id
 
+    def _get_sign_token_params(self):
+        access_token = self._portal_ensure_token()
+        partner = self.env["res.partner"].create({"name": "Sign Partner"})
+        _hash = self._sign_token(partner.id)
+        token = {"token": access_token}
+        bad_token = {"token": "incorrect token"}
+        sign = {"hash": _hash, "pid": partner.id}
+        bad_sign = {"hash": "incorrect hash", "pid": partner.id}
+        return token, bad_token, sign, bad_sign, partner
```

Comment:
I think this would make more sense on `TestControllerCommon` with the target record as param.

Reply:
I'll do a separate PR for it.